proposals/NNNN-sdl-js-pwa.md

```diff
@@ -0,0 +1,213 @@
+
+This definition should ensure that apps can be approved and verified by SDLC and OEMs without possibility of modifications after approval. Also this set of requirements should ensure compatibility throughout integrators. The final approval process will be part of another proposal.
+
+## Potential downsides
+
+The upside of apps running with a WebEngine is that it comes with an extremely flexible html based user interface and with a very sandboxed runtime environment. This can also be seen as downsides as the responsibility of following driver distraction rules increases and that a sandbox environment is limiting the functionality of apps.
+
+## Impact on existing code
+
+1. Core needs a new transport type to support a WebSocket Server.
+2. The JavaScript library needs a new transport type to support WebSocket Client.
+3. Developer site needs a new field for local apps.
+4. Policy table and server need to support the new App HMI type `OPEN_HMI`.
+
+## Alternatives considered
+
+An additional (alternative) feature considered is to use cloud app transport adapter to inform Core about a new app installation. Originally this concept was part of this proposal. However it was removed from this proposal as the cloud app transport adapter is designed for the concept of Core being a client to connect to apps being the server. Using the cloud app properties of the cloud app transport adapter with this way of local web apps could require changes to the app properties and policies leading to the risk of versioning the policy server.
+
+### High level overview extending cloud transport adapter
+
+1. The OEM store uses `SetCloudAppProperties` using the manifest data.
+  1. The OEM store may choose to set `enabled` parameter to `true` to include the app in the HMI RPC `UpdateAppList`.
+  2. If the HMI already supports listing apps installed by the OEM store, the `enabled` parameter should be `false` to avoid listing the app in `UpdateAppList`.
+  3. For local apps
+    1. "cloudTransportType" will be "ws" (WebSocket) or "wss" (WebSocket-Secure)
+    2. "endpoint" parameter will be omitted. Instead the HMI is responsible to launch local apps.
+  4. If Core doesn't know the app ID it should ask HMI for a policy update.
+     1. Alternatively HMI can call `UpdateSDL` to enforce SDL to peform the policy update.
+2. Policy HMI impl. (incl. SYNCP) should have the ability to use embedded modem to send policy snapshot to the policy server.
+3. All CloudAppProperties RPCs and related structs should also be available through HMI_API
+   1. This allows OEM stores easier communication with SDL.
+4. A new enum called "CoreTransportRole" with "CLIENT" and "SERVER" elements should be added
+   1. CloudAppProperties should have a parameter for "coreTransportRole"
+
+### Launching an app (alternative solution)
+
+If the user taps on an app icon and
+  a) and the app is listed as per `UpdateAppList` (app is set with `enabled=true`)
```

Comment:
This is not formatting as a list, but instead as a paragraph.  Can you please fix?

Reply:
fixed